hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -3175,31 +3175,42 @@ private Pair<String, RegionScannerHolder> newRegionScanner(ScanRequest request,
       // wrapper for the core created RegionScanner
       region.getCoprocessorHost().preScannerOpen(scan);
     }
-    RegionScannerImpl coreScanner = region.getScanner(scan);
-    Shipper shipper = coreScanner;
-    RegionScanner scanner = coreScanner;
+    RegionScanner scanner = null;
     try {
-      if (region.getCoprocessorHost() != null) {
-        scanner = region.getCoprocessorHost().postScannerOpen(scan, scanner);
+      RegionScannerImpl coreScanner = region.getScanner(scan);
```

Comment:
I think the intention here is that, if region.getScanner throws exception, there is no way for us to get the reference of the RegionScannerImpl, so just need to throw the exception out, without closing the scanner as we do not have its reference.